### ISSUE TITLE
No code style formatting file found.

### ISSUE THREAD

**Issue by ecruz165** [OHDSI/WebAPI#2203]
### Expected behavior
Project should have a style guide that can be loaded to IDE to ensure same formatting rules are set before finalizing code.

I suggest the following guide
https://github.com/google/styleguide/blob/gh-pages/eclipse-java-google-style.xml
or similar that can be loaded to ones IDE.

### Actual behavior
None exists

**Comment by chrisknoll** [comment]
Not just that, originally it was 2 space indent, and then we standardized on 1 tab indent. But then some people had 2 tab (!??!) indentation which was flat out wrong, but the differences you are seeing is due to a style change after a few years into development.   The complete fix is to reformat every file which would lead to a huge commit, so I think at this point we're asking people to follow the 1-tab indent convention with any changed line and eventually we'll get to the point that the remainder of changes is small.

I believe @anthonysena investgated using an `editorconfig` file in an attempt to standardize this.   Some of the defaults (IMO) are not idea.  I've found that VSCode actually infers style by checking the file. 

I didn't look at all the settings in your example (there's ~330) but we'd have to ensure that the eclipseIDE settings mirror the other IDEs (like IntelliJ or Netbeans), but this is where editor.config would make things easier.

**Comment by ecruz165** [comment]
Perhaps we can use maven-formatter-plugin that can format based on style-guide file profile in our project base. Or even validate if the file being committed is not formatted according to defined style-guide.

[[formatter-maven-plugin](https://code.revelc.net/formatter-maven-plugin/)](https://code.revelc.net/formatter-maven-plugin/)

**Comment by chrisknoll** [comment]
Yes, but this may force us to re-format the entire codebase which could lead to a pretty large PR with most of the lines added/removed for formatting.   (or else the maven build will fail, right?)

I agree we should have formatting standards, and people have been pretty good about following them, but when the solution involves potentially compromising our build chain, it makes me less enthusiastic.   I feel energy could be spent elsewhere for more value.
